### DISCUSSION TITLE
Is it possible to config which service account a cloud function uses?

### DISCUSSION THREAD

**Discussion by thecannabisapp** [firebase/firebase-tools#6067] [Q&A] [answered]
I have an express app which I've deployed as a gen2 function; by default, this function runs under the default compute service account. I want to limit the permissions on this function by creating a custom IAM service account and configuring the cloud-run container to use this account instead. I know this is possible via the gcloud cli, but I'd have to update my functions, and I'd rather keep it all managed via the firebase cli. I was wondering if this is possible. Thanks :)

**Comment by taeold** [comment, marked as answer]
Yes it's possible! Here's an example:

```
const {onRequest} = require("firebase-functions/v2/https");

exports.hello = onRequest(
  {
    serviceAccount: "your-service-account@"
  },
  (req, res) => {
    res.sendStatus(200);
  });
```

**Reply by thecannabisapp** [reply]
Thank you @taeold
